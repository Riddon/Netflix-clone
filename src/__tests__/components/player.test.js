import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Player } from '../../components';

describe('<Player />', () => {

    it('renders the <Player/> with a bunny video and close clicked on <Overlay/>', () => {
        const { container, getByText, queryByTestId } = render(
            <Player>
                <Player.Button/>
                <Player.Video src="videos/bunny.mp4"/>
            </Player>
        );

        expect(queryByTestId('player')).toBeFalsy();
        fireEvent.click(getByText('Play'));

        expect(queryByTestId('player')).toBeTruthy();
        fireEvent.click(queryByTestId('player'));

        expect(queryByTestId('player')).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Player/> with a bunny video and close clicked on <Close/>', () => {
        const { container, getByText, queryByTestId } = render(
            <Player>
                <Player.Button/>
                <Player.Video src="videos/bunny.mp4"/>
            </Player>
        );

        expect(queryByTestId('player')).toBeFalsy();

        fireEvent.click(getByText('Play'));
        expect(queryByTestId('player')).toBeTruthy();

        fireEvent.click(queryByTestId('player-close'));
        expect(queryByTestId('player')).toBeFalsy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Player/> with a bunny video and close clicked on <Inner/>', () => {
        const { container, getByText, queryByTestId } = render(
            <Player>
                <Player.Button/>
                <Player.Video src="videos/bunny.mp4"/>
            </Player>
        );

        expect(queryByTestId('player')).toBeFalsy();

        fireEvent.click(getByText('Play'));
        expect(queryByTestId('player')).toBeTruthy();

        fireEvent.click(queryByTestId('player-inner'));
        expect(queryByTestId('player')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });
});