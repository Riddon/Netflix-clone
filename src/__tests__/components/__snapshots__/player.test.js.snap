// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders the <Player/> with a bunny video and close clicked on <Close/> 1`] = `
<div
  class="player__Container-sc-1c49yiu-0"
>
  <button
    class="player__Button-sc-1c49yiu-4 ePMIUa"
  >
    Play
  </button>
</div>
`;

exports[`<Player /> renders the <Player/> with a bunny video and close clicked on <Inner/> 1`] = `
<div
  class="player__Container-sc-1c49yiu-0"
>
  <button
    class="player__Button-sc-1c49yiu-4 ePMIUa"
  >
    Play
  </button>
</div>
`;

exports[`<Player /> renders the <Player/> with a bunny video and close clicked on <Overlay/> 1`] = `
<div
  class="player__Container-sc-1c49yiu-0"
>
  <button
    class="player__Button-sc-1c49yiu-4 ePMIUa"
  >
    Play
  </button>
</div>
`;
